sys/signal.c

```diff
@@ -0,0 +1,221 @@
+#include <signal.h>
+#include <thread.h>
+#include <malloc.h>
+#include <stdc.h>
+#include <errno.h>
+#include <sysent.h>
+#include <systm.h>
+
+static MALLOC_DEFINE(sig_pool, "signal handlers pool");
+
+static sighandler_t *signal_default_actions[SIG_LAST] = {
+    [SIGINT] = SIG_TERM,  [SIGILL] = SIG_TERM,  [SIGFPE] = SIG_TERM,
+    [SIGABRT] = SIG_TERM, [SIGSEGV] = SIG_TERM, [SIGKILL] = SIG_TERM,
+    [SIGCHLD] = SIG_IGN,  [SIGUSR1] = SIG_TERM, [SIGUSR2] = SIG_TERM,
+};
+
+static const char *signal_names[SIG_LAST] = {
+    [SIGINT] = "SIGINT",   [SIGILL] = "SIGILL",   [SIGABRT] = "SIGABRT",
+    [SIGFPE] = "SIGFPE",   [SIGSEGV] = "SIGSEGV", [SIGKILL] = "SIGKILL",
+    [SIGTERM] = "SIGTERM", [SIGCHLD] = "SIGCHLD", [SIGUSR1] = "SIGUSR1",
+    [SIGUSR2] = "SIGUSR2",
+};
+
+void sighand_init() {
+  /* TODO: A lot of kernel components have a init function whose only purpose is
```

Comment:
Mechanism similar to [SYSINIT](https://www.freebsd.org/doc/en/books/arch-handbook/sysinit-using.html) could be implemented to achieve it. It's not necessary to introduce it right now, altough if specified in details it could be a nice addition to task list (under category of *wish*).